packages/mgt-chat/src/statefulClient/graph.chat.ts

```diff
@@ -1,37 +1,124 @@
 import { IGraph, prepScopes } from '@microsoft/mgt-element';
 import { Chat, ChatMessage } from '@microsoft/microsoft-graph-types';
 
-export type MessageCollection = {
+/**
+ * Generic collection response from graph
+ */
+type GraphCollection<T = any> = {
   '@odata.nextLink': string;
   nextLink: string;
-  value: ChatMessage[];
+  value: T[];
 };
 
+/**
+ * Object representing a collection of chat messages
+ */
+export type MessageCollection = GraphCollection<ChatMessage>;
+
+/**
+ * Object mapping chat operations to the scopes required to perform them
+ */
+const chatOperationScopes: Record<string, string[]> = {
+  loadChat: ['chat.readbasic'],
+  loadChatMessages: ['chat.read'],
+  sendChatMessage: ['chatmessage.send'],
+  updateChatMessage: ['chat.readwrite'],
+  deleteChatMessage: ['chat.readwrite']
+};
+
+/**
+ * Provides an array of the distinct scopes required for all chat operations
+ */
+export const allChatScopes = Array.from(
+  Object.values(chatOperationScopes).reduce((acc, scopes) => {
+    scopes.forEach(s => acc.add(s));
+    return acc;
+  }, new Set<string>())
```

Comment:
A very good use of the `Array.reduce` method with an initial value. I'll consider this approach for any future refactors 🥳